tests/Unit/Evolution/Systems/GeneralizedHarmonic/Test_UpwindFlux.cpp

```diff
@@ -0,0 +1,297 @@
+#include <utility>
+
+#include "ApparentHorizons/Strahlkorper.hpp"
+#include "ApparentHorizons/StrahlkorperGr.hpp"
+#include "ApparentHorizons/Tags.hpp"  // IWYU pragma: keep
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"  // IWYU pragma: keep
+#include "DataStructures/DataVector.hpp"        // IWYU pragma: keep
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "DataStructures/Tensor/EagerMath/DivideBy.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Characteristics.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Equations.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ComputeSpacetimeQuantities.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
+#include "tests/Unit/NumericalAlgorithms/DiscontinuousGalerkin/NumericalFluxes/TestHelpers.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+#include "tests/Utilities/MakeWithRandomValues.hpp"
+
+// IWYU pragma: no_forward_declare GeneralizedHarmonic::Tags::Pi
+// IWYU pragma: no_forward_declare GeneralizedHarmonic::Tags::Phi
+// IWYU pragma: no_forward_declare GeneralizedHarmonic::Tags::UPsi
+// IWYU pragma: no_forward_declare GeneralizedHarmonic::Tags::UZero
+// IWYU pragma: no_forward_declare GeneralizedHarmonic::Tags::UMinus
+// IWYU pragma: no_forward_declare GeneralizedHarmonic::Tags::UPlus
+// IWYU pragma: no_forward_declare StrahlkorperTags::CartesianCoords
+// IWYU pragma: no_forward_declare StrahlkorperTags::NormalOneForm
+// IWYU pragma: no_forward_declare Tags::CharSpeed
+// IWYU pragma: no_forward_declare Tags::dt
+// IWYU pragma: no_forward_declare Tensor
+// IWYU pragma: no_forward_declare Variables
+
+namespace GeneralizedHarmonic {
+namespace GeneralizedHarmonic_detail {
+template <size_t Dim>
+db::item_type<Tags::CharacteristicFields<Dim, Frame::Inertial>>
+weight_char_fields(
+    const db::item_type<Tags::CharacteristicFields<Dim, Frame::Inertial>>&
+        char_fields_int,
+    const db::item_type<Tags::CharacteristicSpeeds<Dim, Frame::Inertial>>&
+        char_speeds_int,
+    const db::item_type<Tags::CharacteristicFields<Dim, Frame::Inertial>>&
+        char_fields_ext,
+    const db::item_type<Tags::CharacteristicSpeeds<Dim, Frame::Inertial>>&
+        char_speeds_ext) noexcept;
+}  // namespace GeneralizedHarmonic_detail
+}  // namespace GeneralizedHarmonic
+
+namespace {
+// Test GH upwind flux using random fields
+void test_upwind_flux_random(
+    const Strahlkorper<Frame::Inertial>& strahlkorper) noexcept {
```

Comment:
This doesn't need to be a Strahlkörper. For your spatial normal one-forms you can choose random values. You only have to make sure, they are not zero.